### ISSUE TITLE
Autobuild: Add Github Action for dependency update PRs

### ISSUE THREAD

**Issue by hoffie** [jamulussoftware/jamulus#2346]
**Describe the task**

We have several external dependencies which are version-locked for reproducibility and security. We should regularly check those for updates. A Github Action scheduled job (cron-style) could do that. It could automatically submit a PR with the suggested update.

- [ ] Github Actions (`uses: `)
- [ ] Pinned dependencies in the build process (aqt, Qt, JACK), see #2345

**Comment by ann0see** [comment]
Dependabot?

**Comment by hoffie** [comment]
> Dependabot?

Yeah, might be worth a look wrt Github Actions usage. I would not expect it to catch the depds from our build scripts though, unless I'm missing something?

**Comment by ann0see** [comment]
Maybe we can set it up to do that

**Comment by hoffie** [comment]
From https://docs.github.com/en/code-security/supply-chain-security/managing-vulnerabilities-in-your-projects-dependencies/configuring-dependabot-security-updates#supported-repositories

> **[](https://docs.github.com/en/code-security/supply-chain-security/managing-vulnerabilities-in-your-projects-dependencies/configuring-dependabot-security-updates#supported-repositories)Supported repositories**
> Repository contains dependency manifest file from a package ecosystem that GitHub supports: "[Supported package ecosystems](https://docs.github.com/en/github/visualizing-repository-data-with-graphs/about-the-dependency-graph#supported-package-ecosystems)"

So it looks like:
- [x] Github Actions could work.
- [x] pip (for aqt) could work (although we would have to create and use a `pipfile.lock`

That'd mean that the following would require manual work (at least):
- [ ] choco
- [ ] Qt

**Comment by ann0see** [comment]
> The workflow does not handle most of the Android pinnings (haven't looked into it so far).

I think we should list them in this issue.

I just found out that create-dmg is also missing. So macOS and Android probably need some further investigation.

**Comment by ann0see** [comment]
Not sure what is still outstanding here. Tagging for looking at it in the next release.

**Comment by ann0see** [comment]
--> moving to next release

**Comment by pljones** [comment]
This is something an AI Agent should be able to do pretty easily and the result should be pretty easy to verify.

But I'm dropping from 4.0.0 until there's a set of PRs.
